core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -154,6 +155,7 @@ public SqlCodeGenerator(
     @Override
     public void clear() {
         whereClauseParser.clear();
```

Comment:
Estimator could be used for several subqueries so clearing state only here doesn't seem to be enough . 

Example : 

```sql
create table st2 (s1 symbol, s2 symbol, i int, ts timestamp) timestamp(ts) partition by year ;

insert into st2 select x::symbol s1, x::symbol s2, x::int, x::timestamp   from long_sequence(2048);

select * from st2 where s1 != 'a' latest on ts partition by s1, s2
union all 
select * from st2 where s2 in ('b', 'c') and s1 = 'b' latest on ts partition by s1, s2
```

because gaveUp flag is kept between estimate() calls the result for second subuqery is incorrect .

Reply:
Good catch. Fixed in 9b2d0b9a9da2408aa6339c4d49d1dd595ee1cd46